src/renderers/shaders/ShaderChunk/alphatest_fragment.glsl.js

```diff
@@ -2,6 +2,7 @@ export default /* glsl */`
 #ifdef ALPHATEST
 
 	if ( diffuseColor.a < ALPHATEST ) discard;
+	diffuseColor.a = 1.0;
```

Comment:
While glTF materials never use both `.alphaTest > 0` and `.transparent = true`, threejs can — i.e. both a cutoff, and semitransparency. I think this would lose semitransparency in that case. Can't remember if there's a `define` to check for transparency...

Reply:
Hmm, I guess I thought that was a bug. Why would you simultaneously use alpha to represent a mask and blending? Seems like it should be either/or.

Reply:
I haven't personally needed to, and I haven't heard complaints from glTF users, but it's not unheard of:

https://github.com/KhronosGroup/glTF/issues/822#issuecomment-278716773

> You have cutout grass, but really do want to have nice antialiased edges, so you then want both: alpha test for whether the fragment should be discarded (and so not affect the z-buffer ... which can lead to rendering errors) and whether the fragments partially covering a pixel should be blended in...

^this has nothing to do with physically-based rendering, and is (IMO) a better use case for alpha hash or temporal dither, but threejs and glTF don't support either of those now.

Reply:
yeah, plus it won't help much if you cutoff is at 0.5, since half of the blend is discarded anyway. This can be done in a shader, but not how three is set up.

Reply:
True, but a careful choice of the cutoff is par for the course here.